src/modes/ModeEstimation.jl

```diff
@@ -78,39 +78,30 @@ end
 """
     OptimLogDensity{M<:Model,C<:Context,V<:VarInfo}
 
-A struct that stores the log density function of a `DynamicPPL` model.
+A struct that stores the negative log density function of a `DynamicPPL` model.
```

Comment:
Is this change correct? Doesn't it now "store" the actual log density function rather than "negative"?

Though the current docstring is confusing anyways.

Reply:
Well, currently `OptimLogDensity` stores the negative log density function (see, e.g., https://github.com/TuringLang/Turing.jl/blob/9951638916a4372bef31f20c723224d49e1f4124/src/modes/ModeEstimation.jl#L113). This makes sense since basically all optimization libraries expect a minimization problem. So I thought it would be best to keep the same behaviour for the `OptimLogDensity` alias and correct the docstring. I'm not completely happy though since it seems somewhat confusing and possibly cleaner to always return the log density when `LogDensityFunction` is evaluated.

Thus I thought about instead changing the implementations of the tilde pipeline for `OptimizationContext` to return the negative log density values instead (and hence adjust for the negative values there) but I think this would break if someone calls `@addlogprob!` or DynamicPPL.acclogp!! in a model.

Reply:
But I think even though this change is correct it is responsible for the test errors: In the implementation of the Optim API below currently we set everything to negative values manually but when unifying the types we already get the correct gradients etc. So the negative signs are wrong in the Optim implementation I think.